### PR TITLE
Feat/button component

### PR DESCRIPTION
**Pull request by cawo88**
## Proposed changes

This Pull Request updates the Button component with additional props including [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role) attributes. Added some basic unit tests using [Jest](https://create-react-app.dev/docs/running-tests) that is used by Create React App too. 

## Text for the Changelog

Button_example

- Changed Button component with additional props including ARIA attributes. 
- Added some basic unit tests to the Button component using Jest and Enzyme.
- Removed @testing-library packages.
 

### Features

- 🚀 Changed Button component with additional props including ARIA attributes.

### Technical

- 🔧Removed [@testing-library/jest-dom](https://github.com/testing-library/jest-dom), [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro) and [@testing-library/user-event](https://testing-library.com/docs/ecosystem-user-event) utility libraries.
- 🔧Added [enzyme](https://github.com/enzymejs/enzyme), enzyme-adapter-react-16 and react-test-renderer.
- 🔧Added `react/jsx-sort-props` rules to EsLint config.
- 🗎  Changed readme file to include `yarn storybook` script.

## Ticket

- [Create a button component for the Pattern Library](https://app.asana.com/0/1154394619859001/1169698267529770)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -17,5 +16,15 @@
...
+  "rules": {
+    "react/jsx-sort-props": [
+      "error",
+      {
+        "callbacksLast": true,
+        "shorthandLast": true,
+        "noSortAlphabetically": false
+      }
+    ]
   }
```
Comment (a14m) [inline .eslintrc:29]: I would prefer to have a unanimous linting configuration across different projects...
so maybe we can start adapting one as a source of truth ([we can start with the template one](https://github.com/infographicsgroup/contentful-starter/blob/master/.eslintrc.js)) and get other projects to adapt the same...
If you think that it's not enough, then we can also add the extra rules there.

I would also prefer to have the .eslint/tsconfig.json and other configuration file options to be alphabetically sorted (makes it easier to check via just looking at the file)

File: .prettierrc [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "printWidth": 120,
+  "printWidth": 80,
   "singleQuote": false,
   "tabWidth": 2,
   "trailingComma": "all"
```
Comment (a14m) [inline .prettierrc:5]: would also prefer to have unanimous prettier configurations across different projects (but that's a personal preference in having all the organization code following the same rules than each having their own set of rules)

**Comment by cawo88** [comment]
> Can we remove the `package-lock.json`? we are already using `yarn.lock` so it's better to keep using just one package manager... and other than minor configuration preferences, LGTM

done

File: .eslintrc (hunk 1 above)
Comment (cawo88) [reply .eslintrc]: Done, I left Cypress linting for now as we are not using Cypress in this project yet.

**Pull request merged**